libsigner/src/messages.rs

```diff
@@ -362,7 +363,11 @@ impl StacksMessageCodec for SignerMessage {
                 SignerMessage::BlockResponse(block_response)
             }
             SignerMessageTypePrefix::Transactions => {
-                let transactions = read_next::<Vec<StacksTransaction>, _>(fd)?;
+                // I don't think these messages are stored on the blockchain, so `StacksEpochId::latest()` should be fine
```

Comment:
Yes the are (or rather will be). Please use the current system epoch.

Reply:
Specifically, you should filter out transactions that would not be mineable in the current epoch, and keep the rest.  These are transactions that the signers can require the miner to include in their blocks, so the signer should not be able to compel the miner to mine invalid transactions.